### ISSUE TITLE
add an `uninstall` CLI option

### ISSUE THREAD

**Issue by 2bndy5** [cpp-linter/clang-tools-pip#26]
I've been _really_ wanting this feature for testing purposes, but it will also be useful to users if we automate this ourselves.

- [x] can be specified on CLI via `-u` or `--uninstall`
- [x] remove the symlink if it targets the uninstalling binary
- [x] perform uninstall action before doing install action. This will allow users to easily re-install binary if they so desire.
- [x] uninstalling a non-existent binary should fail silently (don't raise an `FileNotFoundError`)

**Comment by shenxianpeng** [comment]
> - [ ] perform uninstall action before doing install action. This will allow users to easily re-install binary if they so desire.

If users want to re-install binary, they should perform uninstall action first, then perform install action. 
Doing install action should check `Path(directory, f"{tool_name}-{version}{suffix}").exists()` instead of doing uninstall action.

**Comment by 2bndy5** [comment]
The idea for that statement is to let users re-install with 1 command:
```
clang-tools -u 12 -i 12
```
But it should work the same as
```
clang-tools -i 12 -u 12
```

Did you have a different usage in mind?

**Comment by shenxianpeng** [comment]
Got it, the usage with re-install makes sense.

**Comment by 2bndy5** [comment]
Note to self: Let the readme declare that this pkg only deals with binaries that it installs (not dynamically linked binaries from LLVM releases).

**Comment by 2bndy5** [comment]
I noticed that there's a default in the `--install` arg. If a user only intends to uninstall a version and doesn't specify the `-i` arg, then they may unexpectedly get v13 installed.
```
clang-tools -u 10
```
This will uninstall v10 but also install v13 (even though it wasn't specified).

---
I think the `--install` and `--uninstall` commands would be better served as [subsequent commands](https://docs.python.org/3/library/argparse.html#sub-commands):
```
clang-tools install 13
clang-tools uninstall 10
```
This approach is similar to `pip` and `git` CLI tools.

**Comment by 2bndy5** [comment]
Alternatively, I could just remove the default value for `-i` and avoid refactoring the project for sub-commands. This idea would also require a prompt telling the user that nothing was done when nothing was specified for `-i` and `-u` args.

**Comment by shenxianpeng** [comment]
> Alternatively, I could just remove the default value for `-i` and avoid refactoring the project for sub-commands. This idea would also require a prompt telling the user that nothing was done when nothing was specified for `-i` and `-u` args.

I like this idea

**Comment by 2bndy5** [comment]
> > Alternatively, I could just remove the default value for `-i` and avoid refactoring the project for sub-commands. This idea would also require a prompt telling the user that nothing was done when nothing was specified for `-i` and `-u` args.
> 
> I like this idea

Great!, I already have that solution stashed locally. I can roll that into #29 if that's ok with you.

**Comment by shenxianpeng** [comment]
Of course, it's ok with me.
